### DISCUSSION TITLE
Adding a link to a device.

### DISCUSSION THREAD

**Discussion by fanynek86** [Ylianst/MeshCentral#6499] [Ideas]
Hello gentlemen!
I would like to ask... Is it possible to add a field with a link. Like, for example, a link to noVNC or RDP, so I would add a link to RealVNC, for example, or a link to anywhere.

With BR
František Finda

**Comment by si458** [comment]
You can use the notes section for this,
The notes supports full markdown
set `showNotesPanel: true` under domains in your config.json
Restart meshcentral
Then when u click notes, add ur notes there, then save
And a new window pops up with ur notes and when u click link it's a link :)
Note, the link doesn't open in a new tab! Need to fix that!

**Comment by fanynek86** [comment]
Thank you for your quick response. Your markdown doesn't work with URIs. Here's an example where the link works: 

https://markdown-it.github.io/

Example: [Real VNC](com.realvnc.vncviewer.connect://ID)

When to open Real VNC Viewer and contact the station.

**Reply by si458** [reply]
try this `[link description](http://www.google.com)`

https://www.markdownguide.org/cheat-sheet/

**Comment by fanynek86** [comment]
We don't understand each other, I'm sorry... I have no idea why the link in the example disappeared here.

Example:
[Real VNC](com.realvnc.vncviewer.connect://192.168.1.1)

**Reply by fanynek86** [reply]
I'm so sorry everyone but I'm not able to post so I have to add a picture.
![md](https://github.com/user-attachments/assets/47c6d2b1-4528-452b-9c82-e8355747c388)

**Reply by si458** [reply]
right ok i understand now!
so the must be a bug with the markedjs javascript we use to render the markdown,
will investigate!

edit: its the DOMPurify thats the issue as this sanitises all text before being passed to MarkedJS

**Comment by fanynek86** [comment]
I'm sorry, but the entry I tried to add always disappeared. Thank you for your patience.

I tried using quotation marks, apostrophes, etc. to eliminate disinfection, but in vain :-D

**Reply by si458** [reply]
its the DOMPurify thats the issue as this sanitises all text after markedjs has rendered it into html code for your browser
DomPurify basically expects the A href to be a URL like http:// or https://
so just working out what options it needs to fix it for you!

**Reply by si458** [reply]
this isnt going to be an easy fix - https://github.com/cure53/DOMPurify/tree/main/demos#hook-to-white-list-safe-uri-schemes-link
we would have to list every type of uri scheme to be allowed!
in your case `com.realvnc.vncviewer.connect` is the scheme

if you are trying to use VNC and you have the meshagent installed on the device in question, 
you can use the Web-VNC option which is VNC in the web browser!

or even if you create a local-only group, you can add a device with its IP address, then click Web-VNC there
(so long as meshcentral can access the ip address directly)

**Comment by fanynek86** [comment]
Can't add to "config.json" which uri it will accept?

**Reply by si458** [reply]
OOOO GOOD IDEA!!!! 💡

**Comment by fanynek86** [comment]
May I be of some help or may I ask what should I add to the "config.json" to use this functionality?

**Reply by si458** [reply]
i havent implemented the feature yet,
its on my road map to try sort this month
so just hang fire and ill look into it 👍

**Comment by fanynek86** [comment]
I don't want to push in any way, I want to ask them if this topic is still in process or if I should get some help.

**Reply by si458** [reply]
is the notes panel still not working for yourself?

**Reply by fanynek86** [reply]
Last time we talked, this "allowlist" would have to be added to config.json. I have added lines in the notes, when hovering the mouse, a hand appears, as if they are clickable, but after clicking, nothing happens. I was giving an example with vnc.

**Reply by si458** [reply]
oh right ok! yes no i didnt implement this feature as got side tracked if i remember!
please open a github issue as enhancement and ask in their for the feature to be added and we will add it when we get chance :)
